src/operations.cpp

```diff
@@ -1546,12 +1552,30 @@ path read_symlink(const path& p, system::error_code* ec)
   if (!error(::DeviceIoControl(h.handle, FSCTL_GET_REPARSE_POINT,
         0, 0, info.buf, sizeof(info), &sz, 0) == 0 ? BOOST_ERRNO : 0, p, ec,
         "boost::filesystem::read_symlink" ))
+  {
+    const wchar_t* buffer;
+    size_t offset, len;
+    if (info.rdb.ReparseTag == IO_REPARSE_TAG_MOUNT_POINT)
+    {
+      buffer = info.rdb.MountPointReparseBuffer.PathBuffer;
+      offset = info.rdb.MountPointReparseBuffer.PrintNameOffset;
+      len = info.rdb.MountPointReparseBuffer.PrintNameLength;
+    } else if (info.rdb.ReparseTag == IO_REPARSE_TAG_SYMLINK)
```

Comment:
(nitpick) Add newline before `else`. Ditto in the following code.

Or even better, convert to a `switch`/`case` block.

Reply:
Good point! 👍